lib/logstash/inputs/cloudwatch_logs.rb

```diff
@@ -158,6 +184,54 @@ def priority_of(group)
     @priority.index(group) || -1
   end
 
+  public
+  def create_ddb_table
+    @logger.debug("Creating dynamodb table: #{@dynamodb_sincedb_table}")
```

Comment:
Is there any previous-art on official logstash plugins using dynamo for state?

I see the value in supporting dynamo, but am not sure I like the auto-creation behaviour.  I think I would prefer we error and boot, and require the user to correctly create the table.

We could provide documentation on the correct schema, but not doing it automatically would prevent any issues arising from misconfiguration which may cause it to go back and re-read histories, which could be unintended.

Reply:
I've not see anything else use DDB, so just went with what worked for us.

Regarding autocreation - thinking analogously: a sincedb filesystem file based record is also created automatically? I like autocreation because it makes my deployment simpler, especially as we may end up with quite a few of these.

However, I see your point too - perhaps we should have a configuration parameter which switches between autocreate and error behaviours?

Reply:
I'm good with a configuration to change the behaviour - defaulting to auto-create will probably reduce number of issues logged, but also allow more control for the situation I was concerned about.